tests/0150-telemetry_mock.c

```diff
@@ -37,12 +37,21 @@ typedef struct {
                                    expected diff*/
         int broker_id;          /* Broker id of request. */
 } rd_kafka_telemetry_expected_request_t;
-
+static void test_clear_request_list(rd_kafka_mock_request_t **requests,
+                                    size_t request_cnt) {
+        size_t i;
+        for (i = 0; i < request_cnt; i++) {
+                rd_kafka_mock_request_destroy(requests[i]);
+        }
+        rd_free(requests);
+}
 static void test_telemetry_check_protocol_request_times(
-    rd_kafka_mock_request_t **requests_actual,
-    size_t actual_cnt,
+    rd_kafka_mock_cluster_t *mcluster,
     rd_kafka_telemetry_expected_request_t *requests_expected,
     size_t expected_cnt) {
+        size_t actual_cnt;
+        rd_kafka_mock_request_t **requests_actual =
+            rd_kafka_mock_get_requests(mcluster, &actual_cnt);
```

Comment:
Why was this changed? We either should also remove `rd_kafka_mock_get_requests` in the methods or keep this as it is